### PR TITLE
fix: location details on new booker

### LINKED ISSUE #9695: [CAL-2013] new booker: in-person got removed, now shows "in person"
<@U01NL7HM0SV> did the location in person get removed from the new booker for a reason? it just says in person (organizer location) now?

[Slack Message](https://calendso.slack.com/archives/C01NXDSRY9W/p1687363793134999?thread_ts=1687363793.134999&cid=C01NXDSRY9W)

<sub>[CAL-2013](https://linear.app/calcom/issue/CAL-2013/new-booker-in-person-got-removed-now-shows-in-person)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Fixes https://github.com/calcom/cal.com/issues/9695

Screenshot 2023-06-22 at 3 02 44 PM

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Go to event type settings and change location to organiser phone number or organiser address and enter a value in the second field and check show on booking page input.

- Go to the booking page and see if the value is displayed in the location section.

### CODE REVIEW

File: packages/features/bookings/components/event-meta/Locations.tsx [hunk 1]
```diff
@@ -9,13 +9,31 @@ import { EventMetaBlock } from "./Details";
...
+
   if (!locations?.length) return null;
 
+  const getLocationToDisplay = (location: PublicEvent["locations"][number]) => {
+    const locationType = getEventLocationType(location.type);
+    const defaultLocationLabel = t(locationType?.label ?? "");
+
+    switch (locationType?.label) {
+      case "in_person":
+        return location?.address ?? defaultLocationLabel;
```
Comment (Udit-takkar) [inline packages/features/bookings/components/event-meta/Locations.tsx:21]: if "show of booking page" is not checked we only display the defaultLocationLabel which in this case would be In Person (Organizer Address)

**Comment by Udit-takkar** [comment]
@PeerRich i don't remember exactly but previous booking page did display the link meeting url when added but the new booker only displays text "Link Meeting".

File: packages/features/bookings/components/event-meta/Locations.tsx (hunk 1 above)
Comment (sean-brydon) [reply packages/features/bookings/components/event-meta/Locations.tsx:21]: nice job

Comment (hariombalhara) [reply packages/features/bookings/components/event-meta/Locations.tsx:21]: @Udit-takkar We should use the same logic as in https://github.com/calcom/cal.com/blob/main/apps/web/components/booking/AvailableEventLocations.tsx#L26 

There seems to be something else as well that's missing.

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Left a comment on reusing existing logic which might be handling more things

**Review by sean-brydon — DISMISSED** [review]
Working locally - LGTM

File: packages/features/bookings/components/event-meta/Locations.tsx (hunk 1 above)
Comment (Hybes) [reply packages/features/bookings/components/event-meta/Locations.tsx:21]: it used to show a tooltip on hover too, for long named locations that overflow

Comment (PeerRich) [reply packages/features/bookings/components/event-meta/Locations.tsx:21]: @hariombalhara wanna add the patch yourself or wait for @Udit-takkar ?

Comment (Udit-takkar) [reply packages/features/bookings/components/event-meta/Locations.tsx]: @PeerRich  fixed now

**Comment by hariombalhara** [comment]
Static Link Video Apps like Campfire aren't showing their actual URL in Tooltip it seems.

_Not a blocker though, let's fix it in a followup._
image

**Comment by hariombalhara** [comment]
Also, unrelated to this PR but I noticed that the Tooltip says on Booking Form also that 'Select on the next step'. It only makes sense on date/time selection page. cc @Jaibles

**Comment by hariombalhara** [comment]
image
For a team even when Organizer's default conferencing app is chosen, it's not translated. It was handled in that code, we need to abstract that out as well.

**Review by hariombalhara — APPROVED** [review]

**Pull request merged**